elasticsearch/src/http/transport.rs

```diff
@@ -422,6 +461,66 @@ impl Transport {
         if let Some(q) = query_string {
             request_builder = request_builder.query(q);
         }
+        Ok(request_builder)
+    }
+
+    /// Creates an asynchronous request that can be awaited
+    pub async fn send<B, Q>(
+        &self,
+        method: Method,
+        path: &str,
+        headers: HeaderMap,
+        query_string: Option<&Q>,
+        body: Option<B>,
+        timeout: Option<Duration>,
+    ) -> Result<Response, Error>
+    where
+        B: Body,
+        Q: Serialize + ?Sized,
+    {
+        // Threads will execute against old connection pool during reseed
+        if self.conn_pool.reseedable() {
+            let connection = self.conn_pool.next();
+            let scheme = &connection.url.scheme();
+            // Build node info request
+            let node_request = self.request_builder(
+                &connection,
+                Method::Get,
+                "_nodes/_all/http",
+                headers.clone(),
+                None::<&Q>,
+                None::<B>,
+                timeout,
+            )?;
+            let resp = node_request.send().await?;
+            let json: Value = resp.json().await?;
+            let connections: Vec<Connection> = json["nodes"]
+                .as_object()
+                .unwrap()
+                .iter()
+                .map(|h| {
+                    let url = format!(
+                        "{}://{}",
+                        scheme,
+                        h.1["http"]["publish_address"].as_str().unwrap()
```

Comment:
I _think_ `publish_address` can not be returned in _some_ cases. I'll check if this is still the case for 7.x, but the .NET client checks `publish_address` first and if null or empty, gets the first `bound_address`.

In addition, the format can include the fully qualified domain name in front of the IP address, proceeded by a `/`. See the SniffParser regex and method in the .Net client:

https://github.com/elastic/elasticsearch-net/blob/dde89a6004014de69a38b8755d82c08d412df130/src/Elasticsearch.Net/Responses/Sniff/SniffResponse.cs#L17-L34